eth_dummy_phy/examples/simple/main/emac2emac.c

```diff
@@ -0,0 +1,297 @@
+    /* we can get the ethernet driver handle from event data */
+    esp_eth_handle_t eth_handle = *(esp_eth_handle_t *)event_data;
+
+    switch (event_id) {
+    case ETHERNET_EVENT_CONNECTED:
+        esp_eth_ioctl(eth_handle, ETH_CMD_G_MAC_ADDR, mac_addr);
+        ESP_LOGI(TAG, "Ethernet Link Up");
+        ESP_LOGI(TAG, "Ethernet HW Addr %02x:%02x:%02x:%02x:%02x:%02x",
+                 mac_addr[0], mac_addr[1], mac_addr[2], mac_addr[3], mac_addr[4], mac_addr[5]);
+#if CONFIG_EXAMPLE_DHCP_SERVER_EN
+        esp_netif_dhcps_start(esp_netif_get_handle_from_ifkey("ETH_DEF"));
+#endif
+        break;
+    case ETHERNET_EVENT_DISCONNECTED:
+        ESP_LOGI(TAG, "Ethernet Link Down");
+        break;
+    case ETHERNET_EVENT_START:
+        ESP_LOGI(TAG, "Ethernet Started");
+        break;
+    case ETHERNET_EVENT_STOP:
+        ESP_LOGI(TAG, "Ethernet Stopped");
+        break;
+    default:
+        break;
+    }
+}
+
+/** Event handler for IP_EVENT_ETH_GOT_IP */
+static void got_ip_event_handler(void *arg, esp_event_base_t event_base,
+                                 int32_t event_id, void *event_data)
+{
+    ip_event_got_ip_t *event = (ip_event_got_ip_t *) event_data;
+    const esp_netif_ip_info_t *ip_info = &event->ip_info;
+
+    ESP_LOGI(TAG, "Ethernet Got IP Address");
+    ESP_LOGI(TAG, "~~~~~~~~~~~");
+    ESP_LOGI(TAG, "ETHIP:" IPSTR, IP2STR(&ip_info->ip));
+    ESP_LOGI(TAG, "ETHMASK:" IPSTR, IP2STR(&ip_info->netmask));
+    ESP_LOGI(TAG, "ETHGW:" IPSTR, IP2STR(&ip_info->gw));
+    ESP_LOGI(TAG, "~~~~~~~~~~~");
+
+#if !CONFIG_EXAMPLE_DHCP_SERVER_EN
+    ping_start(ip_info);
+#endif
+}
+
+#if CONFIG_EXAMPLE_RMII_CLK_SOURCE_DEV
+IRAM_ATTR static void gpio_isr_handler(void *arg)
+{
+    BaseType_t high_task_wakeup = pdFALSE;
+    TaskHandle_t task_handle = (TaskHandle_t)arg;
+
+    vTaskNotifyGiveFromISR(task_handle, &high_task_wakeup);
+    if (high_task_wakeup != pdFALSE) {
+        portYIELD_FROM_ISR();
+    }
+}
+#endif // CONFIG_EXAMPLE_RMII_CLK_SOURCE_DEV
+
+void app_main(void)
+{
+#if CONFIG_EXAMPLE_RMII_CLK_SOURCE_DEV
+    // ESP32 device which is RMII CLK source needs to wait with its Ethernet initialization for the "RMII CLK
+    // Sink Device" since the RMII CLK input pin (GPIO0) is also used as a boot strap pin. If the "RMII CLK Source Device" didn't wait,
+    // the "RMII CLK Sink Device" could boot into incorrect mode.
+    esp_rom_gpio_pad_select_gpio(EMAC_CLK_OUT_180_GPIO);
+    gpio_set_pull_mode(EMAC_CLK_OUT_180_GPIO, GPIO_FLOATING); // to not affect GPIO0 (so the Sink Device could be flashed)
+    gpio_install_isr_service(0);
+    esp_rom_gpio_pad_select_gpio(CONFIG_EXAMPLE_CLK_SINK_READY_GPIO);
+    gpio_set_direction(CONFIG_EXAMPLE_CLK_SINK_READY_GPIO, GPIO_MODE_INPUT);
+    gpio_set_pull_mode(CONFIG_EXAMPLE_CLK_SINK_READY_GPIO, GPIO_PULLDOWN_ONLY);
+    gpio_set_intr_type(CONFIG_EXAMPLE_CLK_SINK_READY_GPIO, GPIO_INTR_ANYEDGE);
+    gpio_intr_enable(CONFIG_EXAMPLE_CLK_SINK_READY_GPIO);
```

Comment:
because the "ready gpio" is working as a general pupose IO, we can use `gpio_config` function to simplify the above function calls.

Reply:
I would move the comment which is currently present under `#if CONFIG_EXAMPLE_RMII_CLK_SOURCE_DEV` above that statement.